src/Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -690,33 +690,42 @@ public bool TryParseInterface(string intf, [NotNullWhen(true)] out IReadOnlyList
     }
 
     /// <inheritdoc/>
-    public bool HasRemoteAccess(IPAddress remoteIP)
+    public RemoteAccessPolicyResult ShouldAllowServerAccess(IPAddress remoteIP)
     {
         var config = _configurationManager.GetNetworkConfiguration();
+        if (IsInLocalNetwork(remoteIP))
+        {
+            return RemoteAccessPolicyResult.Allow;
+        }
+
         if (config.EnableRemoteAccess)
```

Comment:
Do an early return here just like you did for the LAN check, makes the code easier to follow.